lib/core/src/sdk.rs

```diff
@@ -1399,76 +1400,105 @@ impl LiquidSdk {
-                            let drain_fees_sat = self
-                                .estimate_drain_tx_fee(None, Some(&lbtc_address))
-                                .await?;
-                            let drain_amount_sat =
-                                get_info_res.wallet_info.balance_sat - drain_fees_sat;
-                            let payment_destination = SendDestination::LiquidAddress {
-                                address_data: LiquidAddressData {
-                                    address: lbtc_address,
-                                    asset_id: Some(asset_id.clone()),
-                                    amount: None,
-                                    amount_sat: Some(drain_amount_sat),
-                                    network: self.config.network.into(),
-                                    label: None,
-                                    message: None,
-                                },
-                                bip353_address: None,
-                            };
-                            (drain_amount_sat, Some(drain_fees_sat), payment_destination)
-                        }
-                        (Some(lbtc_address), _) => {
-                            // The BOLT11 invoice has an MRH but no drain is requested,
-                            // so we calculate the fees of a direct Liquid transaction
-                            let fees_sat = self
-                                .estimate_onchain_tx_or_drain_tx_fee(
-                                    invoice_amount_sat,
-                                    &lbtc_address,
-                                    &asset_id,
-                                )
-                                .await?;
-                            (
-                                invoice_amount_sat,
-                                Some(fees_sat),
-                                SendDestination::Bolt11 {
-                                    invoice,
-                                    bip353_address: None,
-                                },
-                            )
-                        }
-                        (None, _) => {
-                            // The BOLT11 invoice has no MRH, so we calculate the fees using a swap
-                            let boltz_fees_total = lbtc_pair.fees.total(invoice_amount_sat);
-                            let user_lockup_amount_sat = invoice_amount_sat + boltz_fees_total;
-                            let lockup_fees_sat = self
-                                .estimate_lockup_tx_or_drain_tx_fee(user_lockup_amount_sat)
-                                .await?;
-                            let fees_sat = boltz_fees_total + lockup_fees_sat;
-                            (
+                (receiver_amount_sat, fees_sat) = match (mrh_address.clone(), req.amount.clone()) {
+                    (Some(lbtc_address), Some(PayAmount::Drain)) => {
```

Comment:
I am not sure we should support drain with bolt11. It seems just something the implementation let us do atm but needs to be abstracted. Is there a specific case we are targeting here or just because we can?

Reply:
It's to handle the LNURL-pay case, where we receive an invoice but we intend to drain to the MRH address